google/auth/aws.py

```diff
@@ -353,6 +353,7 @@ def __init__(
         token_url,
         credential_source=None,
         service_account_impersonation_url=None,
+        service_account_impersonation_options={},
```

Comment:
Sorry I missed this earlier, it seems you have hit a bug. This will create a shared dictionary in all calls, this is not your intent correct?

 https://docs.python-guide.org/writing/gotchas/#mutable-default-arguments

Reply:
Here is an example in a Python3 shell:

```
❯ python
Python 3.10.4 (main, Jul 19 2022, 20:41:22) [GCC 11.3.0] on linux
Type "help", "copyright", "credits" or "license" for more information.
>>> def foo(x={}):
...     x
KeyboardInterrupt
>>> def foo(x, d={}):
...
KeyboardInterrupt
>>> def foo(x, y, d={}):
...     d[x] = y
...
>>> foo("x", "y")
>>>
KeyboardInterrupt
>>> def foo(x, y, d={}):
...     d[x] = y
...     e
KeyboardInterrupt
>>> def foo(x, y, d={}):
...     d[x] = y
...     return d
...
>>> foo('x', 'y')
{'x': 'y'}
>>> foo('z', 'g')
{'x': 'y', 'z': 'g'}
>>>
```

Reply:
Agreed to follow up with a fix for this